runtime/tracing/tracing-core/common/src/aws/smithy/kotlin/runtime/tracing/KotlinLoggingTraceProbe.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.tracing
+
+import mu.KLogger
+import mu.KotlinLogging
+
+/**
+ * A [TraceProbe] that logs events to the [kotlin-logging](https://github.com/MicroUtils/kotlin-logging) library.
+ * kotlin-logging requires configuration (e.g., of Slf4j on JVM) before events sent to the probe will appear in logs.
+ */
+public object KotlinLoggingTraceProbe : TraceProbe {
```

Comment:
correctness: `public`?

Reply:
`public` was intentional. If users want to customize the tracer used in their clients they'll still need access to this probe:

```kotlin
val s3 = S3Client {
    tracer = DefaultTracer(KotlinLoggingTraceProbe, "CustomRootPrefix-")
}
```

Users may also want to use `KotlinLoggingTraceProbe` in a multiplexing or splitting trace probe. I think `public` is correct in this case.

Reply:
I would probably just call it a `LoggingTraceProbe` then. You could mention that for JVM it is an SLF4J compatible logger but I wouldn't mention kotlin-logging specifically anywhere.